Tensile/KernelWriter.py

```diff
@@ -1163,16 +1224,18 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
             waitLocalWrite = 1
           else:
             waitLocalWrite = -1
-          waitLocalRead  = 1 if isResetLroIter else 0
 
         else: # not isResetLroIter
           waitGlobalRead = 1 if u==0 and kernel["PrefetchGlobalRead"] and kernel["PrefetchLocalRead"] else -1
           waitLocalWrite = -1
-          waitLocalRead  = 1 if kernel["PrefetchLocalRead"] else 0
 
         if self.enable["Wait"]:
+          dataAtIter = u//self.numIterPerCoalescedRead - kernel["PrefetchLocalRead"]%kernel["LoopIters"]
+          numReadsIter = min(u+1,kernel["LoopIters"]//self.numIterPerCoalescedRead - kernel["PrefetchLocalRead"]%kernel["LoopIters"])
+          numPrefetchIter = ((u+1)//(kernel["LoopIters"]-kernel["PrefetchLocalRead"]%kernel["LoopIters"]))*((u+1)-(kernel["LoopIters"]-kernel["PrefetchLocalRead"]%kernel["LoopIters"])) if kernel["PrefetchGlobalRead"] else 0
+          skipReadsIter = numReadsIter + numPrefetchIter - dataAtIter - 1
           waitCode = self.wait(kernel, tensorParametersA, tensorParametersB, \
-              waitGlobalRead, waitLocalWrite, waitLocalRead, \
+              waitGlobalRead, waitLocalWrite, skipReadsIter, \
```

Comment:
here skipReadsIter, however, L1185 would wait for all? and make something like s_waitcnt lgkmcnt(0) followed by s_waitcnt lgkmcnt(8)

Reply:
L1185 calculate s_waitcnt for localWrites
L1238 calculate s_waitcnt for localReads, we wait data reads from LDS to register.

we schedule localReads first then schedule localWrites, so we'll see
 ds_read   x4 // read from lds[0]
 ds_write  x8 // write to lds[1]
 s_waitcnt lgkmcnt(0) // wait for writing to lds[1] done
 s_barrier
 s_waitcnt lgkmcnt(8) // it skip 8 localWrites instruction
 ds_read  // read from lds[1] for next loop

this leave us opportunity to schedule like
 ds_write  x8
 ds_read   x4
 s_waitcnt lgkmcnt(4)
 s_barrier
 s_waitcnt lgkmcnt(0)
 ds_read  // read from lds[1] for next loop

Reply:
will this feature allow us do MT256x256x64 for fp16?

Reply:
@ramjana LDS is enough to do 256x256x64 for fp16, but vgprC + tempVgpr will exceed vgpr limit.
@rosenrodt is developing to reduce vgprC usage in his local branch.

Reply:
256x256 requires 256 accRegister. if you use xdlops c/d register calculation for ARCHvgpr allocation(), kernel should able get maxVgprs.  may be thats what we need for ARCHvgpr allocation()